packages/mobx/src/types/observablemap.ts

```diff
@@ -90,11 +90,12 @@ export type IObservableMapInitialValues<K = any, V = any> =
 // just extend Map? See also https://gist.github.com/nestharus/13b4d74f2ef4a2f4357dbd3fc23c1e54
 // But: https://github.com/mobxjs/mobx/issues/1556
 export class ObservableMap<K = any, V = any>
-    implements Map<K, V>, IInterceptable<IMapWillChange<K, V>>, IListenable {
+    implements Map<K, V>, IInterceptable<IMapWillChange<K, V>>, IListenable
+{
     [$mobx] = ObservableMapMarker
-    data_: Map<K, ObservableValue<V>>
-    hasMap_: Map<K, ObservableValue<boolean>> // hasMap, not hashMap >-).
-    keysAtom_: IAtom
+    data_!: Map<K, ObservableValue<V>>
```

Comment:
Will take a look at that. I wonder whether `"useDefineForClassFields": true` has any effect on this, since it replaces assigments with the `Object.defineProperty` calls.

Reply:
Actually
```ts
class Foo {
 prop!;
}
// transpiles to
"use strict";
class Foo {
    constructor() {
        Object.defineProperty(this, "prop", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
    }
}
```
So I quess it doesn't matter?

Reply:
Yes you're right, I still have 7 year old micro-optimizations against ES5 in my head when I wrote first versions 😅. Feel free to merge!